packages/garbo-lib/src/wanderer/freefight.ts

```diff
@@ -3,76 +3,125 @@ import {
+      return yrRate * options.itemValue(drop.drop);
+    });
+  return (
+    itemDrop +
+    meatDrop +
+    bofaValue(options, m) +
+    cookbookbatQuestValue(options, location, m)
+  );
+}
+
+function freeKillValue(
   location: Location,
   forceItemDrops: boolean,
   options: WandererFactoryOptions,
-) {
+  canForceMonster: boolean,
+): FreeKillValue {
   const badAttributes = ["LUCKY", "ULTRARARE", "BOSS"];
   const rates = appearanceRates(location);
   const monsters = getMonsters(location).filter(
     (m) =>
       !badAttributes.some((s) => m.attributes.includes(s)) && rates[m.name] > 0,
   );
 
-  const canDuplicate =
-    SourceTerminal.have() && SourceTerminal.duplicateUsesRemaining() > 0;
-  const canMctwist = have($item`pro skateboard`) && !get("_epicMcTwistUsed");
-  const possibleDuplicateFactor =
-    2 ** [canDuplicate, canMctwist].filter(Boolean).length;
   if (monsters.length === 0) {
-    return 0;
-  } else {
-    return (
-      sum(monsters, (m) => {
-        const items = itemDropsArray(m).filter((drop) =>
-          ["", "n"].includes(drop.type),
-        );
-        const duplicateFactor = !m.attributes.includes("NOCOPY")
-          ? possibleDuplicateFactor
-          : 1;
+    return { value: 0, forcedMonster: $monster`none` };
+  }
 
-        // TODO: this should consider unbuffed meat drop and unbuffed item drop, probably
-        const meatDrop = clamp((m.minMeat + m.maxMeat) / 2, 0, 1000);
-        const itemDrop =
-          duplicateFactor *
-          sum(items, (drop) => {
-            const yrRate =
-              (drop.type === "" && forceItemDrops ? 100 : drop.rate) / 100;
-            return yrRate * options.itemValue(drop.drop);
-          });
-        return itemDrop + meatDrop + bofaValue(options, m);
-      }) / monsters.length
-    );
+  const monsterValues = monsters.map((m) => {
+    return {
+      value: valueMonster(location, m, forceItemDrops, options),
+      forcedMonster: m,
+    };
+  });
+
+  const targetList = [
+    {
+      value: sum(monsterValues, (v) => v.value) / monsters.length,
+      forcedMonster: $monster`none`,
+    },
+  ];
+
+  if (canForceMonster) {
+    targetList.push(maxBy(monsterValues, (v) => v.value));
   }
+
+  return maxBy(targetList, (l) => l.value);
 }
 
 function monsterValues(
   forceItemDrops: boolean,
   options: WandererFactoryOptions,
-): Map<Location, number> {
-  const values = new Map<Location, number>();
+  canForceMonster: boolean,
+): Map<Location, FreeKillValue> {
+  const values = new Map<Location, FreeKillValue>();
   for (const location of Location.all().filter(
     (l) => canAdventureOrUnlock(l) && !underwater(l),
   )) {
-    values.set(
+    const freeKillValuation = freeKillValue(
       location,
-      averageYrValue(location, forceItemDrops, options) +
-        options.freeFightExtraValue(location),
+      forceItemDrops,
+      options,
+      canForceMonster && PeridotOfPeril.canImperil(location),
```

Comment:
I'm just gonna PR libram to make canImperil require the peridot, I think that's the most sensible behavior for that function